internal/exec/stages/disks/luks.go

```diff
@@ -338,6 +338,14 @@ func (s *stage) createLuks(config types.Config) error {
 			}
 			delete(s.State.LuksPersistKeyFiles, luks.Name)
 		}
+
+		// It's best to wait here for the /dev/disk/by-*/X entries to be
+		// (re)created, not only for other parts of the initramfs but
+		// also because s.waitOnDevices() can still race with udev's
+		// disk entry recreation.
+		if err := s.waitForUdev(devAlias); err != nil {
```

Comment:
Thanks for having a look, I've now added the same call there. In both cases the LUKS device is opened and I think it makes sense to do the same waitForUdev call, it's just that the format step gets skipped. Not sure what you mean with closing?

Reply:
Or thinking about it again, I guess it was fine not having the waitForUdev call in the reuse case. After all, the purpose here was to account for changes from the formatting, not the appearance of new entries from the opening.

Reply:
The opening won't cause reappearances for other parts of the initramfs. Internal usage at a later stage does waitOnDevices. So what's left would be the option to wait for all entries coming from the opening of the LUKS device node for other parts of the initramfs but I believe those would have to have their waiters anyway, regardless whether it would be about things inside a LUKS device or on slow plain disks etc. In summary I think that's out of scope and the focus here should be to prevent any disappearances of entries after formatting a drive due to a race with udev.